test/EVMHost.cpp

```diff
@@ -174,10 +178,14 @@ void EVMHost::newTransactionFrame()
 {
 	// Clear EIP-2929 account access indicator
 	recorded_account_accesses.clear();
-	// Clear EIP-2929 storage access indicator
 	for (auto& [address, account]: accounts)
 		for (auto& [slot, value]: account.storage)
+		{
+			// Clear EIP-2929 storage access indicator
 			value.access_status = EVMC_ACCESS_COLD;
+			// Mark storage values for proper refund calculation.
+			value.original = value.current;
```

Comment:
This is an important change needed for correct calculation. The new system has three variables: original (value prior to the transaction), previous (... value in the transaction), current (... value in the transaction).